packages/core/src/modules/vc/data-integrity/W3cJsonLdCredentialService.ts

```diff
@@ -339,12 +339,23 @@ export class W3cJsonLdCredentialService {
     agentContext: AgentContext,
     verificationMethod: string
   ): Promise<Key> {
-    const documentLoader = this.w3cCredentialsModuleConfig.documentLoader(agentContext)
-    const verificationMethodObject = await documentLoader(verificationMethod)
-    const verificationMethodClass = JsonTransformer.fromJSON(verificationMethodObject.document, VerificationMethod)
-
-    const key = getKeyFromVerificationMethod(verificationMethodClass)
-    return key
+    if (!verificationMethod.startsWith('did:peer')) {
```

Comment:
What about using `isValidPeerDid`?

Reply:
I have updated this to just check it if starts with `did:` as I am uncertain if it will always be a `did:peer`. I have left the split in the else branch as it is only needed there.